pkg/resources/grant_helpers_internal_test.go

```diff
@@ -7,52 +7,52 @@ import (
 func TestSplitGrantID(t *testing.T) {
 	// Vanilla
 	id := "database_name|schema|view_name|privilege"
-	db, schema, view, priv, err := splitGrantID(id)
+	grant, err := grantIDFromString(id)
 	if err != nil {
 		t.Error(err)
 	}
-	if db != "database_name" {
-		t.Errorf("Expected db to be database_name, got %v", db)
+	if grant.ResourceName != "database_name" {
+		t.Errorf("Expected db to be database_name, got %v", grant.ResourceName)
 	}
-	if schema != "schema" {
-		t.Errorf("Expected schema to be schema, got %v", schema)
+	if grant.SchemaName != "schema" {
+		t.Errorf("Expected schema to be schema, got %v", grant.SchemaName)
 	}
-	if view != "view_name" {
-		t.Errorf("Expected view to be view_name, got %v", view)
+	if grant.ViewName != "view_name" {
+		t.Errorf("Expected view to be view_name, got %v", grant.ViewName)
 	}
-	if priv != "privilege" {
-		t.Errorf("Expected priv to be privilege, got %v", priv)
+	if grant.Privilege != "privilege" {
+		t.Errorf("Expected priv to be privilege, got %v", grant.Privilege)
 	}
 
 	// No view
 	id = "database_name|||privilege"
-	db, schema, view, priv, err = splitGrantID(id)
+	grant, err = grantIDFromString(id)
 	if err != nil {
 		t.Error(err)
 	}
-	if db != "database_name" {
-		t.Errorf("Expected db to be database_name, got %v", db)
+	if grant.ResourceName != "database_name" {
+		t.Errorf("Expected db to be database_name, got %v", grant.ResourceName)
 	}
-	if schema != "" {
-		t.Errorf("Expected schema to be blank, got %v", schema)
+	if grant.SchemaName != "" {
+		t.Errorf("Expected schema to be blank, got %v", grant.SchemaName)
 	}
-	if view != "" {
-		t.Errorf("Expected view to be blank, got %v", view)
+	if grant.ViewName != "" {
+		t.Errorf("Expected view to be blank, got %v", grant.ViewName)
 	}
-	if priv != "privilege" {
-		t.Errorf("Expected priv to be privilege, got %v", priv)
+	if grant.Privilege != "privilege" {
+		t.Errorf("Expected priv to be privilege, got %v", grant.Privilege)
 	}
 
 	// Bad ID
 	id = "database|name-privilege"
-	_, _, _, _, err = splitGrantID(id)
+	_, err = grantIDFromString(id)
```

Comment:
I would also try and do an assert on the type, contents, or message of the error. That way you can double check it is the error you are expecting